src/client.ts

```diff
@@ -19,103 +25,179 @@ export class APIClient {
+      return await this.getJson(endpoint, options);
+    } catch (error) {
+      if ([429, 503, 504].includes(error.status) && retries < 5) {
+        // TODO: add delay before retry (sp - 1/2022)
+        return await this.getJsonWithRetries(endpoint, options, ++retries);
+      } else {
+        console.error(error);
+
+        throw new IntegrationProviderAPIError({
+          cause: error,
+          endpoint,
+          status: error.status,
+          statusText: error.statusText,
+        });
+      }
+    }
+  }
+
   /**
-   * Iterates each group resource in the provider.
-   *
-   * @param iteratee receives each resource to produce entities/relationships
+   * Uses nodejs https.get method to make HTTP requests.
+   * Parses JSON result, returning the data.
+   * @param endpoint
+   * @param options
+   * @returns Promise
    */
-  public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
-  ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
-          {
-            id: 'acme-user-1',
-          },
-        ],
-      },
-    ];
-
-    for (const group of groups) {
-      await iteratee(group);
-    }
+  private getJson(
+    endpoint: string,
+    options: https.RequestOptions,
+  ): Promise<any> {
+    return new Promise((resolve, reject) => {
+      let dataString = '';
+
+      const request = https.get(endpoint, options, (response) => {
+        response.on('error', (error) => {
+          reject({
+            status: 500,
+            statusText: error.message,
+          });
+        });
+
+        response.on('data', (chunk: string) => {
+          dataString += chunk;
+        });
+
+        response.on('end', () => {
+          const result = safeJsonParse(dataString);
+          if (response.statusCode === 200) {
+            resolve({
+              status: 200,
+              data: result.data?.data,
+            });
+          } else {
+            reject({
+              status: response.statusCode,
+              statusText: result.data?.message ?? 'An error occurred.',
+            });
+          }
+        });
+      });
+
+      request.on('error', (error) => {
+        reject({
+          status: 500,
+          statusText: error.message,
+        });
+      });
+    });
+  }
+}
+
+/**
+ * Safely converts string to js object or returns error if string is not valid json.
+ * @param dataString stringified json
+ * @returns {data?: any, error?: Error}
+ */
+function safeJsonParse(dataString: string): {
+  data?: any;
```

Comment:
should have a type other than any that is specific to signalsciences.

Reply:
Ended up removing this method